asset_dashboard/views.py

```diff
@@ -265,6 +270,64 @@ def form_valid(self, form):
             return super().form_invalid(form)
 
 
+class AssetAddEditView(LoginRequiredMixin, TemplateView):
+    template_name = 'asset_dashboard/asset_create_update.html'
+    component = 'js/components/maps/update.js'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        project = Project.objects.get(id=self.kwargs['pk'])
+        context['project'] = project
+
+        phases = Phase.objects.filter(project=project).values(
+            'id',
+            'estimated_bid_quarter',
+            'phase_type',
+            'status'
+        )
+        context['props'] = {
+            'phases': list(phases)
+        }
+
+        existing_geometries = LocalAsset.objects.filter(phase__project=project)
+
+        if existing_geometries.exists():
+            context['props'].update({
+                'existing_geoms': serialize('geojson', existing_geometries, geometry_field='geom')
+            })
+
+        if self.request.GET.get('q'):
+            paginated_results, search_results = search_assets(self.request)
+            context['search_results'] = paginated_results
+
+            # Send back the searched geojson, not paginated
```

Comment:
This endpoint is returning a paginated list to show the search results in a table. And it also returns the geojson from the search. I thought the table needed to be paginated so that a user could click through the table. I can implement the pagination in javascript if you think that's redundant/not worth the efficiency to paginate since we're already sending back an entire geojson object? 

This also would be the only reason not to do DRF, since we're serving an html page with a table. If we did do the DRF, then we'd need to serve the list of search results as json and then render the table in react. We'll also be creating another table on this page, the table that lists all of the assets for a project (not yet implemented).

So we need a view that serves the searched assets via both geojson and as a table, plus a list of already saved assets in another table on that page. The view also needs to accept form data (or is that the only endpoint where you're suggesting we use DRF?). With those requirements, is it the appropriate to contain all that logic in one view or separate across multiple views?

Here's the wireframe for this page to give you a better if you haven't seen it: https://app.moqups.com/oyzaWsCDbsZ7xXShLNvzgbpzpBoQ8D1s/edit/page/ae22b960d